### ISSUE TITLE
Acquiring Tokens for Dashboard login is broken on Kubernetes 1.26

### ISSUE THREAD

**Issue by thatguyatgithub** [kubernetes/dashboard#7848]
### What happened?

Documentation seems to have not been up to date, at least not for Kubernetes 1.26, where user token creation is instructed to be requested as:

```
kubectl -n kubernetes-dashboard create token admin-user
```

### What did you expect to happen?

But the since at least 1.24 the `create token` cli is deprecated:

```
> kubectl -n kubernetes-dashboard create token admin-user                                                                                                           
Error: must specify one of -f and -k

error: unknown command "token admin-user"
See 'kubectl create -h' for help and examples
```

Leaving the setup with no actual secrets/tokens to use
```
> kubectl get serviceaccount -n kubernetes-dashboard                                                                                                          130 ↵ 
NAME                   SECRETS   AGE
admin                  0         156m
default                0         157m
kubernetes-dashboard   0         157m
```

### How can we reproduce it (as minimally and precisely as possible)?

fresh out of 1.26.x kubernetes setup

### Kubernetes Dashboard version

Any

### Kubernetes version

1.26.4

**Comment by haroonb** [comment]
Hello I have a similar issue, that no secrets are created after the setup
I installed via helm and created the SA and CRB using this documentation. https://github.com/kubernetes/dashboard/blob/master/docs/user/access-control/README.md#bearer-token
I can see the token with `kubectl -n kubernetes-dashboard create token admin-user`, but no secrets are created.
```
$ helm list -A && kubectl get secrets  -n kubernetes-dashboard && kubectl get po  -n kubernetes-dashboard && kubectl get serviceaccounts -n kubernetes-dashboard && kubectl get clusterrolebindings.rbac.authorization.k8s.io -n kubernetes-dashboard
NAME                    NAMESPACE               REVISION        UPDATED                                 STATUS          CHART                           APP VERSION
cert-manager            cert-manager            1               2023-06-14 12:21:06.434969392 +0000 UTC deployed        cert-manager-v1.7.1             v1.7.1
kubernetes-dashboard    kubernetes-dashboard    1               2023-06-14 12:49:42.939969721 +0000 UTC deployed        kubernetes-dashboard-6.0.8      v2.7.0
NAME                                         TYPE                 DATA   AGE
kubernetes-dashboard-certs                   Opaque               0      7m18s
kubernetes-dashboard-csrf                    Opaque               1      7m18s
kubernetes-dashboard-key-holder              Opaque               2      7m18s
sh.helm.release.v1.kubernetes-dashboard.v1   helm.sh/release.v1   1      7m18s
NAME                                    READY   STATUS    RESTARTS   AGE
kubernetes-dashboard-64bd57954b-ztfts   1/1     Running   0          7m18s
NAME                   SECRETS   AGE
admin-user             0         3m25s
default                0         35m
kubernetes-dashboard   0         7m18s
NAME                                                   ROLE                                                                               AGE
admin-user                                             ClusterRole/cluster-admin                                                          3m25s
cephfs-csi-nodeplugin                                  ClusterRole/cephfs-csi-nodeplugin                                                  40m
cephfs-csi-provisioner-role                            ClusterRole/cephfs-external-provisioner-runner                                     40m
cert-manager-cainjector                                ClusterRole/cert-manager-cainjector                                                35m
cert-manager-controller-approve:cert-manager-io        ClusterRole/cert-manager-controller-approve:cert-manager-io                        35m
cert-manager-controller-certificates                   ClusterRole/cert-manager-controller-certificates                                   35m
cert-manager-controller-certificatesigningrequests     ClusterRole/cert-manager-controller-certificatesigningrequests                     35m
cert-manager-controller-challenges                     ClusterRole/cert-manager-controller-challenges                                     35m
cert-manager-controller-clusterissuers                 ClusterRole/cert-manager-controller-clusterissuers                                 35m
cert-manager-controller-ingress-shim                   ClusterRole/cert-manager-controller-ingress-shim                                   35m
cert-manager-controller-issuers                        ClusterRole/cert-manager-controller-issuers                                        35m
cert-manager-controller-orders                         ClusterRole/cert-manager-controller-orders                                         35m
cert-manager-webhook:subjectaccessreviews              ClusterRole/cert-manager-webhook:subjectaccessreviews                              35m
cluster-admin                                          ClusterRole/cluster-admin                                                          42m
flannel                                                ClusterRole/flannel                                                                42m
kubeadm:get-nodes                                      ClusterRole/kubeadm:get-nodes                                                      42m
kubeadm:kubelet-bootstrap                              ClusterRole/system:node-bootstrapper                                               42m
kubeadm:node-autoapprove-bootstrap                     ClusterRole/system:certificates.k8s.io:certificatesigningrequests:nodeclient       42m
kubeadm:node-autoapprove-certificate-rotation          ClusterRole/system:certificates.k8s.io:certificatesigningrequests:selfnodeclient   42m
kubeadm:node-proxier                                   ClusterRole/system:node-proxier                                                    42m
kubernetes-dashboard-metrics                           ClusterRole/kubernetes-dashboard-metrics                                           7m18s
rbd-csi-nodeplugin                                     ClusterRole/rbd-csi-nodeplugin                                                     40m
rbd-csi-provisioner-role                               ClusterRole/rbd-external-provisioner-runner                                        40m
rook-ceph-global                                       ClusterRole/rook-ceph-global                                                       40m
rook-ceph-mgr-cluster                                  ClusterRole/rook-ceph-mgr-cluster                                                  40m
rook-ceph-object-bucket                                ClusterRole/rook-ceph-object-bucket                                                40m
rook-ceph-osd                                          ClusterRole/rook-ceph-osd                                                          40m
rook-ceph-system                                       ClusterRole/rook-ceph-system                                                       40m
rook-ceph-system-psp                                   ClusterRole/psp:rook                                                               40m
rook-csi-cephfs-plugin-sa-psp                          ClusterRole/psp:rook                                                               40m
rook-csi-cephfs-provisioner-sa-psp                     ClusterRole/psp:rook                                                               40m
rook-csi-rbd-plugin-sa-psp                             ClusterRole/psp:rook                                                               40m
rook-csi-rbd-provisioner-sa-psp                        ClusterRole/psp:rook                                                               40m
system:basic-user                                      ClusterRole/system:basic-user                                                      42m
system:controller:attachdetach-controller              ClusterRole/system:controller:attachdetach-controller                              42m
system:controller:certificate-controller               ClusterRole/system:controller:certificate-controller                               42m
system:controller:clusterrole-aggregation-controller   ClusterRole/system:controller:clusterrole-aggregation-controller                   42m
system:controller:cronjob-controller                   ClusterRole/system:controller:cronjob-controller                                   42m
system:controller:daemon-set-controller                ClusterRole/system:controller:daemon-set-controller                                42m
system:controller:deployment-controller                ClusterRole/system:controller:deployment-controller                                42m
system:controller:disruption-controller                ClusterRole/system:controller:disruption-controller                                42m
system:controller:endpoint-controller                  ClusterRole/system:controller:endpoint-controller                                  42m
system:controller:endpointslice-controller             ClusterRole/system:controller:endpointslice-controller                             42m
system:controller:endpointslicemirroring-controller    ClusterRole/system:controller:endpointslicemirroring-controller                    42m
system:controller:ephemeral-volume-controller          ClusterRole/system:controller:ephemeral-volume-controller                          42m
system:controller:expand-controller                    ClusterRole/system:controller:expand-controller                                    42m
system:controller:generic-garbage-collector            ClusterRole/system:controller:generic-garbage-collector                            42m
system:controller:horizontal-pod-autoscaler            ClusterRole/system:controller:horizontal-pod-autoscaler                            42m
system:controller:job-controller                       ClusterRole/system:controller:job-controller                                       42m
system:controller:namespace-controller                 ClusterRole/system:controller:namespace-controller                                 42m
system:controller:node-controller                      ClusterRole/system:controller:node-controller                                      42m
system:controller:persistent-volume-binder             ClusterRole/system:controller:persistent-volume-binder                             42m
system:controller:pod-garbage-collector                ClusterRole/system:controller:pod-garbage-collector                                42m
system:controller:pv-protection-controller             ClusterRole/system:controller:pv-protection-controller                             42m
system:controller:pvc-protection-controller            ClusterRole/system:controller:pvc-protection-controller                            42m
system:controller:replicaset-controller                ClusterRole/system:controller:replicaset-controller                                42m
system:controller:replication-controller               ClusterRole/system:controller:replication-controller                               42m
system:controller:resourcequota-controller             ClusterRole/system:controller:resourcequota-controller                             42m
system:controller:root-ca-cert-publisher               ClusterRole/system:controller:root-ca-cert-publisher                               42m
system:controller:route-controller                     ClusterRole/system:controller:route-controller                                     42m
system:controller:service-account-controller           ClusterRole/system:controller:service-account-controller                           42m
system:controller:service-controller                   ClusterRole/system:controller:service-controller                                   42m
system:controller:statefulset-controller               ClusterRole/system:controller:statefulset-controller                               42m
system:controller:ttl-after-finished-controller        ClusterRole/system:controller:ttl-after-finished-controller                        42m
system:controller:ttl-controller                       ClusterRole/system:controller:ttl-controller                                       42m
system:coredns                                         ClusterRole/system:coredns                                                         42m
system:discovery                                       ClusterRole/system:discovery                                                       42m
system:kube-controller-manager                         ClusterRole/system:kube-controller-manager                                         42m
system:kube-dns                                        ClusterRole/system:kube-dns                                                        42m
system:kube-scheduler                                  ClusterRole/system:kube-scheduler                                                  42m
system:monitoring                                      ClusterRole/system:monitoring                                                      42m
system:node                                            ClusterRole/system:node                                                            42m
system:node-proxier                                    ClusterRole/system:node-proxier                                                    42m
system:public-info-viewer                              ClusterRole/system:public-info-viewer                                              42m
system:service-account-issuer-discovery                ClusterRole/system:service-account-issuer-discovery                                42m
system:volume-scheduler                                ClusterRole/system:volume-scheduler                                                42m
traefik-ingress-controller                             ClusterRole/traefik-ingress-controller                                             35m
traefik-ingress-controller-ext                         ClusterRole/traefik-ingress-controller                                             35m
traefik-ingress-controller-int                         ClusterRole/traefik-ingress-controller                                             35m
```

**Comment by floreks** [comment]
`kubectl create token xxx` command is not supposed to create any secrets. It was added to provide a temporary token that can be used to authenticate against the API server.

@thatguyatgithub 
It looks like your `kubectl` might be outdated. I do not see any information about deprecating this command and it has been available since `kubectl` version `v1.23` or `v1.24` IIRC.

Ref: https://github.com/kubernetes/kubectl/issues/1233 - similar issue
Ref #2: https://github.com/kubernetes/kubectl/blob/master/pkg/cmd/create/create_token.go - still available on the master

/close
